升级 2.17 后 `postFinder.listByCategory()` 报错

### 系统信息

- 外部访问地址: https://2a1e-14-155-63-83.ngrok-free.app
- 启动时间: 2024-06-27 15:45
- 版本: 2.17.0-SNAPSHOT
- 构建时间: 2024-06-27 11:50
- Git Commit: 14e5dcd
- Java: IBM Semeru Runtime Open Edition / 21.0.3+9-LTS
- 数据库: PostgreSQL / 15.4 (Debian 15.4-2.pgdg120+1)
- 操作系统: Linux / 5.15.153.1-microsoft-standard-WSL2
- 已激活主题: [Daisyblocks 1.6.1](https://www.halo.run/store/apps/app-WgWHe)
- 已启动插件:
  - [全视之眼 1.0.0-SNAPSHOT](https://github.com/halo-sigs/plugin-god-mode)
  - [插件快速开始模板 1.0.0-SNAPSHOT](https://github.com/halo-dev/plugin-hellopage)
  - [静态网页服务 1.0.0-alpha.1](https://github.com/ruibaby/plugin-static-pages)
  - [Docsme 1.0.0-alpha.5](https://github.com/f2ccloud/plugin-docsme)
  - [SEO 工具集 1.0.0-SNAPSHOT](https://github.com/f2ccloud/plugin-seo-tools)
  - 联系表单 1.2.0-SNAPSHOT
  - Reflect Halo 微信小程序配套插件 1.0.0-SNAPSHOT
  - [应用市场 1.2.0](https://www.halo.run/store/apps/app-VYJbF)
  - [社交 IAM 认证 1.0.2-SNAPSHOT](https://www.halo.run/store/apps/app-IXZkJ)
  - AI 助手 1.0.0-alpha.2
  - [代码注入增强插件 1.0.0-SNAPSHOT](https://github.com/halo-sigs/plugin-codeInjection)
  - [lightgallery.js 灯箱 1.1.1](https://github.com/halo-sigs/plugin-lightgallery)
  - [文章限制阅读 Pro 1.0.0-SNAPSHOT](https://www.halo.run/store/apps/app-TzRqd)
  - [Markdown / HTML 内容块 1.0.1](https://github.com/halo-dev/plugin-hybrid-edit-block)
  - [WangEditor 1.0.0](https://gitee.com/handsome-npc/halo-wangEditor-plugin)
  - [微信公众号发布 1.1.2-SNAPSHOT](https://github.com/f2ccloud/plugin-platforms-sync)
  - [站点迁移 1.3.0](https://github.com/halo-sigs/plugin-migrate)
  - [StackEdit 1.1.1](https://github.com/halo-sigs/plugin-stackedit)
  - [ByteMD 1.4.0](https://github.com/ruibaby/plugin-bytemd)
  - [图库管理 1.3.0](https://github.com/halo-sigs/plugin-photos)
  - [链接管理 1.3.3](https://github.com/halo-sigs/plugin-links)
  - [LDAP 认证 1.0.0-SNAPSHOT](https://github.com/f2ccloud/plugin-ldap-login)
  - [Sitemap 1.0.2](https://github.com/halo-sigs/plugin-sitemap)
  - [Unsplash 1.0.0](https://github.com/halo-sigs/plugin-unsplash)
  - [reveal.js 幻灯片编辑器 1.0.0-rc.1](https://github.com/wan92hen/plugin-revealjs)
  - [瞬间 1.5.1](https://github.com/halo-sigs/plugin-moments)
  - [OAuth2 认证 1.1.0](https://github.com/halo-sigs/plugin-oauth2)
  - [RSS 1.1.1](https://github.com/halo-sigs/plugin-feed)
  - [评论组件 2.0.1](https://github.com/halo-dev/plugin-comment-widget)
  - [搜索组件 1.4.0](https://github.com/halo-dev/plugin-search-widget)
  - [highlight.js 代码高亮 1.2.0](https://github.com/halo-sigs/plugin-highlightjs)

### 使用的哪种方式运行？

Docker

### 发生了什么？

以下主题代码后台报错：
```
    <div th:with="category=${categoryFinder.getByName(theme.config.blog.category)}">
      <div th:txt="${category}">     
      </div>
    </div>
    <div th:with="posts=${postFinder.listByCategory(1,10,theme.config.blog.category)}">
      <div th:each="post : ${posts}">
        <div th:text="${post.spec.title}"></div>
      </div>
    </div>
```
将 `postFinder.listByCategory` 中的分类名固定后正常：
```
    <div th:with="category=${categoryFinder.getByName(theme.config.blog.category)}">
      <div th:txt="${category}">     
      </div>
    </div>
    <div th:with="posts=${postFinder.listByCategory(1,10,'35')}">
      <div th:each="post : ${posts}">
        <div th:text="${post.spec.title}"></div>
      </div>
    </div>
```

### 相关日志输出

```shell
2024-06-27T16:29:40.482+08:00 ERROR 7 --- [undedElastic-29] a.w.r.e.AbstractErrorWebExceptionHandler : [22161674-1588]  500 Server Error for HTTP GET "/"

org.thymeleaf.exceptions.TemplateInputException: An error happened during template parsing (template: "/root/.halo2/themes/theme-daisyblocks/templates/index.html")
        at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parse(AbstractMarkupTemplateParser.java:241) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
Error has been observed at the following site(s):
        *__checkpoint ⇢ Handler run.halo.app.theme.router.factories.IndexRouteFactory$$Lambda/0x00000000681048c0@42ff95e6 [DispatcherHandler]
        *__checkpoint ⇢ run.halo.app.security.InitializeRedirectionWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.app.security.LoginRequiredForAccessWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.app.security.device.DeviceSessionFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.app.sms.security.SmsLoginAuthenticator [DefaultWebFilterChain]
        *__checkpoint ⇢ AuthorizationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ ExceptionTranslationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ LogoutWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ ServerRequestCacheWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ SecurityContextServerWebExchangeWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ LogoutPageGeneratingWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ AnonymousAuthenticationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
        *__checkpoint ⇢ TotpAuthenticationFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
        *__checkpoint ⇢ RememberMeAuthenticationFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
        *__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ ReactorContextWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ CsrfWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ CorsWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ HttpHeaderWriterWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ ServerWebExchangeReactorContextWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.seo.tools.AdvancedRedirectWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.app.sms.security.SmsLoginAuthenticator [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.social.login.security.SocialAuthenticatorFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.oauth.Oauth2Authenticator [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.seo.tools.CrawlRecordFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.social.login.security.SocialAuthorizationRequestRedirectWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.oauth.Oauth2AuthorizationRequestRedirectWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ cc.ryanc.staticpages.endpoint.RewriteOnNotFoundFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ run.halo.app.webfilter.AdditionalWebFilterChainProxy [DefaultWebFilterChain]
        *__checkpoint ⇢ HTTP GET "/" [ExceptionHandlingWebHandler]
Original Stack Trace:
                at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parse(AbstractMarkupTemplateParser.java:241) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
                at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parseStandalone(AbstractMarkupTemplateParser.java:100) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
                at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:666) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
                at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1103) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
                at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1077) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
                at org.thymeleaf.spring6.SpringWebFluxTemplateEngine.lambda$createFullStream$0(SpringWebFluxTemplateEngine.java:198) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
                at reactor.core.publisher.MonoCreate.subscribe(MonoCreate.java:61) ~[reactor-core-3.6.7.jar:3.6.7]
                at reactor.core.publisher.Mono.subscribe(Mono.java:4568) ~[reactor-core-3.6.7.jar:3.6.7]
                at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:126) ~[reactor-core-3.6.7.jar:3.6.7]
                at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.6.7.jar:3.6.7]
                at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.6.7.jar:3.6.7]
                at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[na:na]
                at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
                at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
                at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
                at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
Caused by: org.attoparser.ParseException: Error during execution of processor 'org.thymeleaf.standard.processor.StandardWithTagProcessor' (template: "index" - line 23, col 10)
        at org.attoparser.MarkupParser.parseDocument(MarkupParser.java:393) ~[attoparser-2.0.7.RELEASE.jar:2.0.7.RELEASE]
        at org.attoparser.MarkupParser.parse(MarkupParser.java:257) ~[attoparser-2.0.7.RELEASE.jar:2.0.7.RELEASE]
        at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parse(AbstractMarkupTemplateParser.java:230) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parseStandalone(AbstractMarkupTemplateParser.java:100) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:666) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1103) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1077) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.spring6.SpringWebFluxTemplateEngine.lambda$createFullStream$0(SpringWebFluxTemplateEngine.java:198) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at reactor.core.publisher.MonoCreate.subscribe(MonoCreate.java:61) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.publisher.Mono.subscribe(Mono.java:4568) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:126) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.6.7.jar:3.6.7]
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[na:na]
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
        at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
Caused by: org.thymeleaf.exceptions.TemplateProcessingException: Error during execution of processor 'org.thymeleaf.standard.processor.StandardWithTagProcessor' (template: "index" - line 23, col 10)
        at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:117) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:1204) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.Model.process(Model.java:282) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1587) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.engine.TemplateHandlerAdapterMarkupHandler.handleOpenElementEnd(TemplateHandlerAdapterMarkupHandler.java:304) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.templateparser.markup.InlinedOutputExpressionMarkupHandler$InlineMarkupAdapterPreProcessorHandler.handleOpenElementEnd(InlinedOutputExpressionMarkupHandler.java:278) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.standard.inline.OutputExpressionInlinePreProcessorHandler.handleOpenElementEnd(OutputExpressionInlinePreProcessorHandler.java:186) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.templateparser.markup.InlinedOutputExpressionMarkupHandler.handleOpenElementEnd(InlinedOutputExpressionMarkupHandler.java:124) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.attoparser.HtmlElement.handleOpenElementEnd(HtmlElement.java:109) ~[attoparser-2.0.7.RELEASE.jar:2.0.7.RELEASE]
        at org.attoparser.HtmlMarkupHandler.handleOpenElementEnd(HtmlMarkupHandler.java:297) ~[attoparser-2.0.7.RELEASE.jar:2.0.7.RELEASE]
        at org.attoparser.MarkupEventProcessorHandler.handleOpenElementEnd(MarkupEventProcessorHandler.java:402) ~[attoparser-2.0.7.RELEASE.jar:2.0.7.RELEASE]
        at org.attoparser.ParsingElementMarkupUtil.parseOpenElement(ParsingElementMarkupUtil.java:159) ~[attoparser-2.0.7.RELEASE.jar:2.0.7.RELEASE]
        at org.attoparser.MarkupParser.parseBuffer(MarkupParser.java:710) ~[attoparser-2.0.7.RELEASE.jar:2.0.7.RELEASE]
        at org.attoparser.MarkupParser.parseDocument(MarkupParser.java:301) ~[attoparser-2.0.7.RELEASE.jar:2.0.7.RELEASE]
        ... 17 common frames omitted
Caused by: java.lang.IllegalArgumentException: No index found for fieldPath [metadata.creationTimeStamp], make sure you have created an index for this field.
        at run.halo.app.extension.index.DefaultIndexer.getIndexEntry(DefaultIndexer.java:183) ~[classes/:na]
        at run.halo.app.extension.index.IndexedQueryEngineImpl.checkIndexForNames(IndexedQueryEngineImpl.java:139) ~[classes/:na]
        at run.halo.app.extension.index.IndexedQueryEngineImpl.doRetrieve(IndexedQueryEngineImpl.java:114) ~[classes/:na]
        at run.halo.app.extension.index.IndexedQueryEngineImpl.retrieveAll(IndexedQueryEngineImpl.java:55) ~[classes/:na]
        at run.halo.app.extension.ReactiveExtensionClientImpl.lambda$listAll$2(ReactiveExtensionClientImpl.java:117) ~[classes/:2.17.0-SNAPSHOT]
        at reactor.core.publisher.MonoSupplier$MonoSupplierSubscription.request(MonoSupplier.java:126) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.request(MonoPeekTerminal.java:139) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onSubscribe(MonoFlatMapMany.java:141) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.onSubscribe(MonoPeekTerminal.java:152) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.publisher.MonoSupplier.subscribe(MonoSupplier.java:48) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.publisher.Mono.subscribe(Mono.java:4568) ~[reactor-core-3.6.7.jar:3.6.7]
        at reactor.core.publisher.Mono.block(Mono.java:1778) ~[reactor-core-3.6.7.jar:3.6.7]
        at run.halo.app.theme.ReactiveSpelVariableExpressionEvaluator.evaluate(ReactiveSpelVariableExpressionEvaluator.java:37) ~[classes/:2.17.0-SNAPSHOT]
        at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:125) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.standard.processor.StandardWithTagProcessor.doProcess(StandardWithTagProcessor.java:93) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
        ... 36 common frames omitted
        Suppressed: java.lang.Exception: #block terminated with an error
                at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:104) ~[reactor-core-3.6.7.jar:3.6.7]
                at reactor.core.publisher.Mono.block(Mono.java:1779) ~[reactor-core-3.6.7.jar:3.6.7]
                ... 44 common frames omitted
```


### 附加信息

_No response_

--- a/application/src/main/java/run/halo/app/theme/finders/impl/PostFinderImpl.java
+++ b/application/src/main/java/run/halo/app/theme/finders/impl/PostFinderImpl.java
@@ -165,7 +165,7 @@ public class PostFinderImpl implements PostFinder {
     private Flux<Category> listChildrenCategories(String categoryName) {
         if (StringUtils.isBlank(categoryName)) {
             return client.listAll(Category.class, new ListOptions(),
-                Sort.by(Sort.Order.asc("metadata.creationTimeStamp"),
+                Sort.by(Sort.Order.asc("metadata.creationTimestamp"),
                     Sort.Order.desc("metadata.name")));
         }
         return categoryService.listChildren(categoryName);

